extensions/omni_schema/src/make_plan.sql

```diff
@@ -0,0 +1,129 @@
+create or replace function make_plan(path text) 
+    returns table
+            (
+                migration_filename  text,
+                migration_statement omni_sql.statement
+            )
+    language plpgsql
+as
+$$
+declare
+    local_conn_info text := 'host=localhost port=' || (select setting from pg_settings where name = 'port') || ' user=' || current_user;
+    current_db_conn text := local_conn_info || ' dbname=' || current_database();
+    plan_db text := 'plan_' || gen_random_uuid();
+    plan_db_conn text := local_conn_info || ' dbname=' || plan_db;
+    migration record;
+    stmt omni_sql.statement;
+    error_message text;
+    hint_message text;
+    extension_record record;
+begin
+    perform dblink(current_db_conn, format('create database %I', plan_db));
+    perform dblink(plan_db_conn, 'create extension omni_schema cascade;');
+
+    -- prepare necessary extensions in the remote database
+    for extension_record in select distinct extension
+               from omni_schema.languages
+               union
+               select distinct file_processor_extension as extension
+               from omni_schema.languages
+               union
+               select distinct processor_extension as extension
+               from omni_schema.auxiliary_tools
+    loop
+            if exists (select from pg_extension where extname = extension_record.extension) then
+                perform dblink(plan_db_conn,
+                               format('create extension if not exists %s cascade', extension_record.extension));
+            end if;
+    end loop;
+
+    for migration in select * from dblink(
+        plan_db_conn,
+        format('select migration_filename, executed_statement, migration_statement, execution_error from omni_schema.assemble_schema(%L, omni_vfs.local_fs(%L))', plan_db_conn, path)
```

Comment:
Would this approach mean we might not be able to ensure the order of migrations?

If a migration failed and proceeded further down migrations and only then went for sources, this would mean we'll come and try to re-apply the incremental migration.

Whereas if I got it right, the algorithm is to try migrations in order and when they error our, try to run sources.

Reply:
It seems to me that the `assemble_schema` order will be preserved. But having it implicit is certainly not a good idea. Perhaps we could output an order from `assemble_schema` at least for now. 

The goal was to have something quick that would not require changes in `assemble_schema`. Later it might be better to just rewrite the logic inside `make_plan`.